test/catalog/catalog.test.ts

```diff
@@ -0,0 +1,202 @@
+                expectedSpecific.values()
+            );
+
+            catalog.merge(testCatalog);
+            assert.deepEqual(catalog, expectedCatalog);
+        });
+
+        it('should throw an exception if there is a duplicate item key', () => {
+            const testGeneric: GenericTypedEntity[] = [genericCoffee];
+            const testSpecific: SpecificTypedEntity[] = [
+                smallChocolateCone,
+                smallVanillaCone,
+                mediumCoffee,
+                zeroMilk,
+            ];
+            const catalog = Catalog.fromEntities(
+                testGeneric.values(),
+                testSpecific.values()
+            );
+
+            expect(catalog.merge.bind(testCatalog)).to.throw(TypeError);
+        });
+    });
+
+    describe('hasPID', () => {
+        it('should return true if PID is found ', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            assert.isTrue(catalog.hasPID(genericConePID));
+        });
+        it('should return false if PID is not found', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            assert.isFalse(catalog.hasPID(genericCoffeePID));
+        });
+    });
+
+    describe('getGeneric', () => {
+        it('should return a GenericTypedEntity if PID is found', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            assert.equal(genericCone, catalog.getGeneric(genericConePID));
+        });
+        it('should throw a TypeError if PID is not found', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            expect(catalog.getGeneric.bind(genericCoffeePID)).to.throw(
+                TypeError
+            );
+        });
+    });
+
+    describe('getGenericForKey', () => {
+        it('should return GenericTypedEntity if key is found', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            assert.equal(
+                genericCone,
+                catalog.getGenericForKey(genericCone.defaultKey)
+            );
+        });
+        it('should throw a TypeError if key is not found', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            expect(
+                catalog.getGenericForKey.bind(genericCoffee.defaultKey)
+            ).to.throw(TypeError);
+        });
+    });
+
+    describe('getGenericMap', () => {
+        it('should return map of PID and generic entities item in a catalog', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            for (const map of catalog.getGenericMap()) {
+                assert.deepEqual([genericConePID, genericCone], map);
+            }
+        });
+    });
+
+    describe('genericEntities', () => {
+        it('should return IterableIterator of generics in a catalog', () => {
+            const catalog = Catalog.fromCatalog(testCatalog);
+            for (const entity of catalog.genericEntities()) {
```

Comment:
Could you count the items and verify that there is only one. Also add a comment explaining what this test is doing. As it stands, the code is checking that every entity is a genericCone, which doesn't seem like tight enough semantics. This test and the uniqueness test will guarantee the correct behavior, but the reader might not know this.

Reply:
Made a change for this, what do you think?